mesonbuild/interpreterbase.py

```diff
@@ -160,8 +182,8 @@ def wrapped(*wrapped_args, **wrapped_kwargs):
 
 class permittedKwargs:
 
-    def __init__(self, permitted):
-        self.permitted = permitted
+    def __init__(self, permitted: Set[str]) -> None:
+        self.permitted = permitted  # type: Set[str]
```

Comment:
you shouldn't need to annotate the `self` assignment, mypy will see that self.permitted as to be a `Set[str]` based on the arguments.

Reply:
For me, it is more obvious what the type of a member variable should be if I annotate it directly. So yes, it is redundant for mypy but it makes the code easier to reason about (at least for me).

As an added benefit, it is harder to accendentaly change the type of the variable because now two lines have to be changed to make mypy happy.